skimage/segmentation/_felzenszwalb_cy.pyx

```diff
@@ -12,50 +12,56 @@ from ..measure._ccomp cimport find_root, join_trees
 from ..util import img_as_float
 
 
-def _felzenszwalb_grey(image, double scale=1, sigma=0.8,
-                       Py_ssize_t min_size=20):
-    """Felzenszwalb's efficient graph based segmentation for a single channel.
+def _felzenszwalb_cython(image, double scale=1, sigma=0.8,
+                         Py_ssize_t min_size=20):
+    """Felzenszwalb's efficient graph based segmentation for 
+    single or multiple channels.
 
-    Produces an oversegmentation of a 2d image using a fast, minimum spanning
-    tree based clustering on the image grid.
+    Produces an oversegmentation of a single or multi-channel image 
+    using a fast, minimum spanning tree based clustering on the image grid.
     The number of produced segments as well as their size can only be
     controlled indirectly through ``scale``. Segment size within an image can
     vary greatly depending on local contrast.
 
     Parameters
     ----------
-    image: ndarray
+    image : (N, M, C) ndarray
         Input image.
-    scale: float, optional (default 1)
+    scale : float, optional (default 1)
         Sets the obervation level. Higher means larger clusters.
-    sigma: float, optional (default 0.8)
+    sigma : float, optional (default 0.8)
         Width of Gaussian smoothing kernel used in preprocessing.
         Larger sigma gives smother segment boundaries.
-    min_size: int, optional (default 20)
+    min_size : int, optional (default 20)
         Minimum component size. Enforced using postprocessing.
 
     Returns
     -------
-    segment_mask: (height, width) ndarray
+    segment_mask : (N, M) ndarray
         Integer mask indicating segment labels.
     """
-    if image.ndim != 2:
-        raise ValueError("This algorithm works only on single-channel 2d"
-                "images. Got image of shape %s" % str(image.shape))
+    if image.ndim != 3:
+        raise ValueError("This algorithm works only on single or " 
+                         "multi-channel 3d images. "
```

Comment:
@stefanv & @soupault : According to the user guide then this should read "this works only on 2D multichannel images", correct? As single channel 2d images would be interpreted as being a 2d array - what the user guide refers to as "2D grayscale".